pkgs/development/haskell-modules/lib/compose.nix

```
@@ -108,6 +108,13 @@ rec {
      of test suites listed in the package description file.
    */
   dontCheck = overrideCabal (drv: { doCheck = false; });
+  /* The dontCheckIf variant sets doCheck = false if the condition
+     applies. In any other case the previously set/default value is used.
+     This prevents accidentally re-enabling tests in a later override.
+   */
+  dontCheckIf = condition: overrideCabal (drv:
+    pkgs.lib.optionalAttrs condition { doCheck = false; }
+  );
```

Maybe you can simplify this to this?
```suggestion
  dontCheckIf = condition: if condition then dontCheck else lib.id;
```